drivers/include/lm75.h

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright (C) 2021 ML!PA Consulting GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_lm75
+ *
+ * @{
+ * @file
+ * @brief       Driver for the LM75 temperature sensor.
+ *
+ * @author      Vitor Batista <vitor.batista@ml-pa.com>
+ *
+ * @}
+ */
+
+#ifndef LM75_H
+#define LM75_H
+
+#include "periph/i2c.h"
+
+/**
+ * @name LM75 return values
+ */
+enum {
+    LM75_SUCCESS = 0,
+    LM75_ERROR_I2C,
+    LM75_ERROR
+};
+
+/* temperature resolution of the LM75 sensors */
+typedef struct {
+    uint16_t    os_res;
+    uint8_t     os_bits;
+    uint16_t    os_mult;
+    uint16_t    temp_res;
+    uint8_t     temp_bits;
+    uint16_t    temp_mult;
+} lm75_temp_resolutions;
+
+/* LM75 configuration struct type */
+typedef struct lm75_params {
+    i2c_t   i2c_bus;
+    uint8_t i2c_addr;
+    uint8_t shutdown_mode;
+    uint8_t tm_mode;
+    uint8_t polarity;
+    uint8_t fault_q;
+    uint8_t conv_rate;
+    uint8_t one_shot_mode;
+    uint8_t device_type;
+    lm75_temp_resolutions lm75_res;
+} lm75_params_t;
+
+/* M75 device descriptor */
+typedef struct lm75 {
+    lm75_params_t	lm75_params;
+} lm75_t;
+
+
+/**
+ * @brief Initialization of the LM75 sensor
+ *
+ * Initializes the sensor according to specific input parameters.
+ *
+ * @param[out] dev        device structure to initialize
+ * @param[in] params      initialization parameters
+ *
+ * @return LM75_SUCCESS, on success
+ * @return LM75_ERROR_I2C, on I2C related error
+ * @return LM75_ERROR, on initialization related error
+ */
+int lm75_init(lm75_t *dev, const lm75_params_t *params);
+
+/**
+ * @brief Temperature values of LM75 sensor
+ *
+ * Reads the sensor temperature values from TEMP_REG.
+ *
+ * @param[out] dev                device structure
+ * @param[out] temperature        buffer where temperature value will be written
```

Comment:
What unit is the temperature in?

Reply:
All temperatures are in degrees Celsius. Should I add that to the documentation?

Reply:
Yes please document the units.
Also doesn't the sensor have sub-°C resolution? We shouldn't just throw that away. Maybe use milli°C instead?

Reply:
Yes, the sensors do have sub ºC temperature values(the LM75A has a resolution of 0.5ºC and the TMP1075 of 0.0625), but we are currently only using sub ºC for reading temperatures and not writing. Should I add that feature then?

Reply:
The API should be general and not prevent users from accessing the full precision of the sensor - and we can easily scale the unit to make that happen without the use of floats.

Reply:
I just notice the description was wrong, and the device temperature read is outputted with full precision. However, when writing the desired OS and THYST temperatures for the device only ºC precision is currently available. Should I change this to allow for more specific OS and HYST temperatures?

Reply:
If the hardware supports it the API should make it possible to use higher precision values.